akka-cluster/src/main/scala/akka/cluster/MembershipState.scala

```diff
@@ -22,10 +22,14 @@ import akka.util.ccompat._
 @ccompatUsedUntil213
 @InternalApi private[akka] object MembershipState {
   import MemberStatus._
-  private val leaderMemberStatus = Set[MemberStatus](Up, Leaving)
+  private val leaderMemberStatus = Set[MemberStatus](Up, Leaving, PreparingForShutdown, ReadyForShutdown)
+  // FIXME should shutting down be added to these?
   private val convergenceMemberStatus = Set[MemberStatus](Up, Leaving)
   val convergenceSkipUnreachableWithMemberStatus = Set[MemberStatus](Down, Exiting)
   val removeUnreachableWithMemberStatus = Set[MemberStatus](Down, Exiting)
+  // If a member hasn't join yet or has already started leaving don't mark it as shutting down
+  val allowedToPrepareToShutdown = Set[MemberStatus](Up, WeaklyUp)
```

Comment:
Maybe we should treat WeaklyUp in same way as Joining?
We are not moving WeaklyUp to Up when preparing for shutdown so maybe those that are WeaklyUp should remain so.

Reply:
The more I look at this I'm convinced that WeaklyUp should be removed here. For example leader selection.